crates/provider/src/provider/call.rs

```diff
@@ -14,19 +14,25 @@ type RunningFut<'req, 'state, T, N, Resp, Output, Map> =
 #[derive(Clone, Debug)]
 struct EthCallParams<'req, 'state, N: Network> {
     data: &'req N::TransactionRequest,
-    block: BlockId,
+    block: Option<BlockId>,
     overrides: Option<&'state StateOverride>,
 }
 
 impl<N: Network> serde::Serialize for EthCallParams<'_, '_, N> {
     fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
-        let len = if self.overrides.is_some() { 3 } else { 2 };
+        let len = 1 + self.block.is_some() as usize + self.overrides.is_some() as usize;
+
         let mut seq = serializer.serialize_seq(Some(len))?;
         seq.serialize_element(&self.data)?;
-        seq.serialize_element(&self.block)?;
+
+        if let Some(block) = self.block {
+            seq.serialize_element(&block)?;
+        }
```

Comment:
here we need to always serialize if overrides are set because overrides must be third argument